### ISSUE TITLE
Using TransactionScope with UseXATransactions=False

### ISSUE THREAD

**Issue by dtkujawski** [mysql-net/MySqlConnector#1404]
I understand from other issues in this repo that using TransactionScope will cause the connector to treat the transactions as distributed, but is it possible to use TransactionScope without using XA transactions?  We have a common middleware which uses TransactionScope for all inbound connections and, at where the middleware sits, it wouldn't have access to call dbContext.Database.BeginTransaction() which is why we use TransactionScope.

This works when we point our system to SQL Server or when we point to Oracle, but when we use MySqlConnector with a connectionString which includes UseXATransactions=false we get the error:

**System.NotSupportedException: 'Multiple simultaneous connections or connections with different connection strings inside the same transaction are not supported when UseXaTransactions=False.'**

Is this something that we can somehow do currently or in the future?

**Comment by bgrainger** [comment]
> is it possible to use TransactionScope without using XA transactions?

Yes, with limitations. To perform this, use `UseXaTransactions = false;` in your connection string. The limitation is that you may only open one `MySqlConnection` within that `TransactionScope`. If you try to open a second connection, you will get the exception you reference in your issue.

How are you wanting `TransactionScope` to work with two different open `MySqlConnection`s if you're not using a distributed transaction? Without participating in a distributed transaction, there's no way to atomically commit (or roll back) the changes on the two different connections.

It seems like the right solution would be to use `UseXaTransactions=true;` (i.e., the default) in your connection string. Is this not possible for you?

OTOH, if you just want the error to go away, you could add `Auto Enlist = false;` to your connection string (see https://mysqlconnector.net/connection-options/#AutoEnlist). This would make MySqlConnector completely ignore the `TransactionScope` (and ensuring data integrity would be entirely your responsibility).

**Comment by bgrainger** [comment]
The other solution is to ensure you don't have more than one concurrently-open `MySqlConnection`. Make sure to `Close` the first one before you open a second one. This will allow the `MySqlConnection` to participate in the `TransactionScope` (without using an XA transaction) and still ensure data integrity (by commiting/aborting the transaction based on how the `TransactionScope` is resolved).
